pyteal/ast/binaryexpr_test.py

```diff
@@ -1385,3 +1385,40 @@ def test_b_ge_invalid():
 
     with pytest.raises(TealTypeError):
         BytesGe(Bytes("base16", "0xFF"), Int(2))
+
+
+def test_extract_uint():
+    for expression, op in (
+        (ExtractUint16, Op.extract_uint16),
+        (ExtractUint32, Op.extract_uint32),
+        (ExtractUint64, Op.extract_uint64),
+    ):
+        args = [
+            Bytes("base16", "0xFFFFFFFFFFFFFFFFFF"),
+            Int(2),
+        ]
+        expr = expression(args[0], args[1])
+        assert expr.type_of() == TealType.uint64
+
+        expected = TealSimpleBlock(
+            [
+                TealOp(args[0], Op.byte, "0xFFFFFFFFFFFFFFFFFF"),
+                TealOp(args[1], Op.int, 2),
+                TealOp(expr, op),
+            ]
+        )
+
+        actual, _ = expr.__teal__(options)
+        actual.addIncoming()
+        actual = TealBlock.NormalizeBlocks(actual)
+
+        assert actual == expected
+
+
+def test_extract_uint_invalid():
```

Comment:
Should there be a test that fails when `Teal4Options` are passed in?

Reply:
Good point, adding this now